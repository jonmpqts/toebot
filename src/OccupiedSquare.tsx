import * as React from 'react';
import * as FontAwesome from 'react-fontawesome';
import * as classNames from 'classnames';
import './Square.css';

interface Props {
    occupiedBy: 'X' | 'O';
}

const OccupiedSquare = ({ occupiedBy }: Props) => {
    const classes = {
        square: true,
    };

    classes[occupiedBy] = true;

    const icon = occupiedBy === 'X' ? 'times' : 'circle-o';

    return (
        <div className={classNames(classes)}>
            <FontAwesome name={icon} />
        </div>
    );
};

export default OccupiedSquare;
