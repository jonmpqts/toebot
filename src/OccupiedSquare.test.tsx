import * as React from 'react';
import * as renderer from 'react-test-renderer';
import OccupiedSquare from './OccupiedSquare';

test('renders first player', () => {
    const component = renderer.create(
        <OccupiedSquare occupiedBy="X" />
    );

    expect(component.toJSON()).toMatchSnapshot();
});

test('renders second player', () => {
    const component = renderer.create(
        <OccupiedSquare occupiedBy="O" />
    );

    expect(component.toJSON()).toMatchSnapshot();
});
