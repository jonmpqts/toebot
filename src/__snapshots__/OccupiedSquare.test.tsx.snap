// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders first player 1`] = `
<div
  className="square X"
>
  <span
    aria-hidden={true}
    className="fa fa-times"
  />
</div>
`;

exports[`renders second player 1`] = `
<div
  className="square O"
>
  <span
    aria-hidden={true}
    className="fa fa-circle-o"
  />
</div>
`;
